### PR TITLE
[XXX] Bump govuk-lint to 3.11.4

### PR DESCRIPTION
**Pull request by benilovj**
### Context
Since the rubocop version bump, the Rails cops have been broken.

### Changes proposed in this pull request
- bump the `govuk-lint` gem to pick up the re-introduction of the Rails cops
- auto-fix and manually fix flagged issues
- disable the `Rails/HasManyOrHasOneDependent` for now to get back to green

### Guidance to review
Is it right to disable the `Rails/HasManyOrHasOneDependent` cop?

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by dankmitchell** [comment]
I think we should add `dependant` to the necessary associations but perhaps another PR or commit if this is going to get too big...

**Comment by benilovj** [comment]
> I think we should add `dependant` to the necessary associations but perhaps another PR or commit if this is going to get too big...

Agreed, let's discuss tomorrow in the office.

File: app/models/course.rb [hunk 1]
```diff
@@ -82,6 +82,7 @@ class Course < ApplicationRecord
            ->(course) { where(provider_code: course.provider.provider_code) },
            foreign_key: :ucas_course_code,
            primary_key: :course_code,
+           inverse_of: 'course',
```
Comment (fofr) [inline app/models/course.rb]: What does this do?

Comment (benilovj) [reply app/models/course.rb]: https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Rails/InverseOf

**Pull request merged**